test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/RepositorySignatureInfoUtilityTests.cs

```diff
@@ -0,0 +1,302 @@
+        }
+
+        [Fact]
+        public void GetSignedPackageVerifierSettings_NullRepoSignatureInfoReturnsCommonSettings()
+        {
+            // Arrange
+            var commonSettings = _defaultSettings;
+
+            // Act
+            var settings = RepositorySignatureInfoUtility.GetSignedPackageVerifierSettings(null, commonSettings);
+
+            // Assert
+            settings.Should().Be(commonSettings);
+        }
+
+        [Fact]
+        public void GetSignedPackageVerifierSettings_RepoSignatureInfoTrueAllSignedClearsAllowUnsignedIfSet()
+        {
+            // Arrange
+            var commonSettings = _defaultSettings;
+            var allSigned = true;
+            var repoSignatureInfo = new RepositorySignatureInfo(allSigned, null);
+
+            // Act
+            var settings = RepositorySignatureInfoUtility.GetSignedPackageVerifierSettings(repoSignatureInfo, commonSettings);
+
+            // Assert
+            settings.AllowUnsigned.Should().BeFalse();
+            settings.AllowIllegal.Should().BeTrue();
+            settings.AllowUntrusted.Should().BeTrue();
+            settings.AllowUntrustedSelfIssuedCertificate.Should().BeTrue();
+            settings.AllowIgnoreTimestamp.Should().BeTrue();
+            settings.AllowNoTimestamp.Should().BeTrue();
+            settings.AllowMultipleTimestamps.Should().BeTrue();
+            settings.AllowUnknownRevocation.Should().BeTrue();
+        }
+
+        [Fact]
+        public void GetSignedPackageVerifierSettings_RepoSignatureInfoTrueAllSignedClearsAllowUnsignedIfNotSet()
+        {
+            // Arrange
+            var commonSettings = _verifyCommandDefaultSettings;
+            var allSigned = true;
+            var repoSignatureInfo = new RepositorySignatureInfo(allSigned, null);
+
+            // Act
+            var settings = RepositorySignatureInfoUtility.GetSignedPackageVerifierSettings(repoSignatureInfo, commonSettings);
+
+            // Assert
+            settings.AllowUnsigned.Should().BeFalse();
+            settings.AllowIllegal.Should().BeFalse();
+            settings.AllowUntrusted.Should().BeFalse();
+            settings.AllowUntrustedSelfIssuedCertificate.Should().BeTrue();
+            settings.AllowIgnoreTimestamp.Should().BeFalse();
+            settings.AllowNoTimestamp.Should().BeTrue();
+            settings.AllowMultipleTimestamps.Should().BeTrue();
+            settings.AllowUnknownRevocation.Should().BeTrue();
+            settings.AllowNoClientCertificateList.Should().BeTrue();
+            settings.AllowNoRepositoryCertificateList.Should().BeFalse();
+        }
+
+        [Fact]
+        public void GetSignedPackageVerifierSettings_RepoSignatureInfoFalseAllSignedDoesNotSetAllowUnsigned()
+        {
+            // Arrange
+            var commonSettings = _defaultSettings;
+            var allSigned = false;
+            var repoSignatureInfo = new RepositorySignatureInfo(allSigned, null);
```

Comment:
Use a named parameter at least for the `null` argument.  This explains the `null` literal.  You can also eliminate the need for the `allSigned` local.  Code becomes more succinct and readable.

    var repoSignatureInfo = new RepositorySignatureInfo(
        allRepositorySigned: false,
        repositoryCertificateInfos: null);